From 77447eddeb946fa7828de2e6020a433e3adf6c61 Mon Sep 17 00:00:00 2001
From: Takamitsu <takamitsu.honda.pv@renesas.com>
Date: Sat, 30 Oct 2021 14:34:03 +0900
Subject: [PATCH] arm64: dts: renesas: r8a779f0: Fix SPI number of ICCOM UIO
 and uio-name

Apply latest MFIS information.
The SPI number was changed start 322, 324, 326.. end 368.
Rename uio-name from 01_xx to 00_xx.

Signed-off-by: Takamitsu Honda <takamitsu.honda.pv@renesas.com>
---
 arch/arm64/boot/dts/renesas/r8a779f0.dtsi | 126 ++++++----------------
 1 file changed, 31 insertions(+), 95 deletions(-)

diff --git a/arch/arm64/boot/dts/renesas/r8a779f0.dtsi b/arch/arm64/boot/dts/renesas/r8a779f0.dtsi
index 931320116f2a..9c304d812627 100644
--- a/arch/arm64/boot/dts/renesas/r8a779f0.dtsi
+++ b/arch/arm64/boot/dts/renesas/r8a779f0.dtsi
@@ -1013,7 +1013,7 @@
 		iccom01 {
 			compatible = "generic-uio";
 			reg = <0 0xe6260000 0 0x10000>;
-			interrupts = <GIC_SPI 321 IRQ_TYPE_LEVEL_HIGH>;
+			interrupts = <GIC_SPI 322 IRQ_TYPE_LEVEL_HIGH>;
 			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
 			linux,uio-name = "irq_iccom_00_01";
 		};
@@ -1021,7 +1021,7 @@
 		iccom02 {
 			compatible = "generic-uio";
 			reg = <0 0xe6260000 0 0x10000>;
-			interrupts = <GIC_SPI 322 IRQ_TYPE_LEVEL_HIGH>;
+			interrupts = <GIC_SPI 324 IRQ_TYPE_LEVEL_HIGH>;
 			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
 			linux,uio-name = "irq_iccom_00_02";
 		};
@@ -1029,7 +1029,7 @@
 		iccom03 {
 			compatible = "generic-uio";
 			reg = <0 0xe6260000 0 0x10000>;
-			interrupts = <GIC_SPI 323 IRQ_TYPE_LEVEL_HIGH>;
+			interrupts = <GIC_SPI 326 IRQ_TYPE_LEVEL_HIGH>;
 			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
 			linux,uio-name = "irq_iccom_00_03";
 		};
@@ -1037,7 +1037,7 @@
 		iccom04 {
 			compatible = "generic-uio";
 			reg = <0 0xe6260000 0 0x10000>;
-			interrupts = <GIC_SPI 324 IRQ_TYPE_LEVEL_HIGH>;
+			interrupts = <GIC_SPI 328 IRQ_TYPE_LEVEL_HIGH>;
 			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
 			linux,uio-name = "irq_iccom_00_04";
 		};
@@ -1045,7 +1045,7 @@
 		iccom05 {
 			compatible = "generic-uio";
 			reg = <0 0xe6260000 0 0x10000>;
-			interrupts = <GIC_SPI 325 IRQ_TYPE_LEVEL_HIGH>;
+			interrupts = <GIC_SPI 330 IRQ_TYPE_LEVEL_HIGH>;
 			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
 			linux,uio-name = "irq_iccom_00_05";
 		};
@@ -1053,7 +1053,7 @@
 		iccom06 {
 			compatible = "generic-uio";
 			reg = <0 0xe6260000 0 0x10000>;
-			interrupts = <GIC_SPI 326 IRQ_TYPE_LEVEL_HIGH>;
+			interrupts = <GIC_SPI 332 IRQ_TYPE_LEVEL_HIGH>;
 			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
 			linux,uio-name = "irq_iccom_00_06";
 		};
@@ -1061,7 +1061,7 @@
 		iccom07 {
 			compatible = "generic-uio";
 			reg = <0 0xe6260000 0 0x10000>;
-			interrupts = <GIC_SPI 327 IRQ_TYPE_LEVEL_HIGH>;
+			interrupts = <GIC_SPI 334 IRQ_TYPE_LEVEL_HIGH>;
 			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
 			linux,uio-name = "irq_iccom_00_07";
 		};
@@ -1069,7 +1069,7 @@
 		iccom08 {
 			compatible = "generic-uio";
 			reg = <0 0xe6260000 0 0x10000>;
-			interrupts = <GIC_SPI 328 IRQ_TYPE_LEVEL_HIGH>;
+			interrupts = <GIC_SPI 336 IRQ_TYPE_LEVEL_HIGH>;
 			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
 			linux,uio-name = "irq_iccom_00_08";
 		};
@@ -1077,7 +1077,7 @@
 		iccom09 {
 			compatible = "generic-uio";
 			reg = <0 0xe6260000 0 0x10000>;
-			interrupts = <GIC_SPI 329 IRQ_TYPE_LEVEL_HIGH>;
+			interrupts = <GIC_SPI 338 IRQ_TYPE_LEVEL_HIGH>;
 			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
 			linux,uio-name = "irq_iccom_00_09";
 		};
@@ -1085,7 +1085,7 @@
 		iccom010 {
 			compatible = "generic-uio";
 			reg = <0 0xe6260000 0 0x10000>;
-			interrupts = <GIC_SPI 330 IRQ_TYPE_LEVEL_HIGH>;
+			interrupts = <GIC_SPI 340 IRQ_TYPE_LEVEL_HIGH>;
 			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
 			linux,uio-name = "irq_iccom_00_10";
 		};
@@ -1094,7 +1094,7 @@
 		iccom011 {
 			compatible = "generic-uio";
 			reg = <0 0xe6260000 0 0x10000>;
-			interrupts = <GIC_SPI 331 IRQ_TYPE_LEVEL_HIGH>;
+			interrupts = <GIC_SPI 342 IRQ_TYPE_LEVEL_HIGH>;
 			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
 			linux,uio-name = "irq_iccom_00_11";
 		};
@@ -1102,7 +1102,7 @@
 		iccom012 {
 			compatible = "generic-uio";
 			reg = <0 0xe6260000 0 0x10000>;
-			interrupts = <GIC_SPI 332 IRQ_TYPE_LEVEL_HIGH>;
+			interrupts = <GIC_SPI 344 IRQ_TYPE_LEVEL_HIGH>;
 			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
 			linux,uio-name = "irq_iccom_00_12";
 		};
@@ -1110,7 +1110,7 @@
 		iccom013 {
 			compatible = "generic-uio";
 			reg = <0 0xe6260000 0 0x10000>;
-			interrupts = <GIC_SPI 333 IRQ_TYPE_LEVEL_HIGH>;
+			interrupts = <GIC_SPI 346 IRQ_TYPE_LEVEL_HIGH>;
 			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
 			linux,uio-name = "irq_iccom_00_13";
 		};
@@ -1118,145 +1118,81 @@
 		iccom014 {
 			compatible = "generic-uio";
 			reg = <0 0xe6260000 0 0x10000>;
-			interrupts = <GIC_SPI 334 IRQ_TYPE_LEVEL_HIGH>;
+			interrupts = <GIC_SPI 348 IRQ_TYPE_LEVEL_HIGH>;
 			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
 			linux,uio-name = "irq_iccom_00_14";
 		};
 
 		iccom015 {
-			compatible = "generic-uio";
-			reg = <0 0xe6260000 0 0x10000>;
-			interrupts = <GIC_SPI 335 IRQ_TYPE_LEVEL_HIGH>;
-			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
-			linux,uio-name = "irq_iccom_00_15";
-		};
-
-		iccom100 {
-			compatible = "generic-uio";
-			reg = <0 0xe6260000 0 0x10000>;
-			interrupts = <GIC_SPI 336 IRQ_TYPE_LEVEL_HIGH>;
-			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
-			linux,uio-name = "irq_iccom_01_00";
-		};
-
-		iccom101 {
-			compatible = "generic-uio";
-			reg = <0 0xe6260000 0 0x10000>;
-			interrupts = <GIC_SPI 338 IRQ_TYPE_LEVEL_HIGH>;
-			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
-			linux,uio-name = "irq_iccom_01_01";
-		};
-
-		iccom102 {
-			compatible = "generic-uio";
-			reg = <0 0xe6260000 0 0x10000>;
-			interrupts = <GIC_SPI 340 IRQ_TYPE_LEVEL_HIGH>;
-			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
-			linux,uio-name = "irq_iccom_01_02";
-		};
-
-		iccom103 {
-			compatible = "generic-uio";
-			reg = <0 0xe6260000 0 0x10000>;
-			interrupts = <GIC_SPI 342 IRQ_TYPE_LEVEL_HIGH>;
-			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
-			linux,uio-name = "irq_iccom_01_03";
-		};
-
-		iccom104 {
-			compatible = "generic-uio";
-			reg = <0 0xe6260000 0 0x10000>;
-			interrupts = <GIC_SPI 344 IRQ_TYPE_LEVEL_HIGH>;
-			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
-			linux,uio-name = "irq_iccom_01_04";
-		};
-
-		iccom105 {
-			compatible = "generic-uio";
-			reg = <0 0xe6260000 0 0x10000>;
-			interrupts = <GIC_SPI 346 IRQ_TYPE_LEVEL_HIGH>;
-			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
-			linux,uio-name = "irq_iccom_01_05";
-		};
-
-		iccom106 {
-			compatible = "generic-uio";
-			reg = <0 0xe6260000 0 0x10000>;
-			interrupts = <GIC_SPI 348 IRQ_TYPE_LEVEL_HIGH>;
-			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
-			linux,uio-name = "irq_iccom_01_06";
-		};
-
-		iccom107 {
 			compatible = "generic-uio";
 			reg = <0 0xe6260000 0 0x10000>;
 			interrupts = <GIC_SPI 350 IRQ_TYPE_LEVEL_HIGH>;
 			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
-			linux,uio-name = "irq_iccom_01_07";
+			linux,uio-name = "irq_iccom_00_15";
 		};
 
-		iccom108 {
+		iccom016 {
 			compatible = "generic-uio";
 			reg = <0 0xe6260000 0 0x10000>;
 			interrupts = <GIC_SPI 354 IRQ_TYPE_LEVEL_HIGH>;
 			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
-			linux,uio-name = "irq_iccom_01_08";
+			linux,uio-name = "irq_iccom_00_16";
 		};
 
-		iccom109 {
+		iccom017 {
 			compatible = "generic-uio";
 			reg = <0 0xe6260000 0 0x10000>;
 			interrupts = <GIC_SPI 356 IRQ_TYPE_LEVEL_HIGH>;
 			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
-			linux,uio-name = "irq_iccom_01_09";
+			linux,uio-name = "irq_iccom_00_17";
 		};
 
-		iccom110 {
+		iccom018 {
 			compatible = "generic-uio";
 			reg = <0 0xe6260000 0 0x10000>;
 			interrupts = <GIC_SPI 358 IRQ_TYPE_LEVEL_HIGH>;
 			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
-			linux,uio-name = "irq_iccom_01_10";
+			linux,uio-name = "irq_iccom_00_18";
 		};
 
-		iccom111 {
+		iccom019 {
 			compatible = "generic-uio";
 			reg = <0 0xe6260000 0 0x10000>;
 			interrupts = <GIC_SPI 360 IRQ_TYPE_LEVEL_HIGH>;
 			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
-			linux,uio-name = "irq_iccom_01_11";
+			linux,uio-name = "irq_iccom_00_19";
 		};
 
-		iccom112 {
+		iccom020 {
 			compatible = "generic-uio";
 			reg = <0 0xe6260000 0 0x10000>;
 			interrupts = <GIC_SPI 362 IRQ_TYPE_LEVEL_HIGH>;
 			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
-			linux,uio-name = "irq_iccom_01_12";
+			linux,uio-name = "irq_iccom_00_20";
 		};
 
-		iccom113 {
+		iccom021 {
 			compatible = "generic-uio";
 			reg = <0 0xe6260000 0 0x10000>;
 			interrupts = <GIC_SPI 364 IRQ_TYPE_LEVEL_HIGH>;
 			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
-			linux,uio-name = "irq_iccom_01_13";
+			linux,uio-name = "irq_iccom_00_21";
 		};
 
-		iccom114 {
+		iccom022 {
 			compatible = "generic-uio";
 			reg = <0 0xe6260000 0 0x10000>;
 			interrupts = <GIC_SPI 366 IRQ_TYPE_LEVEL_HIGH>;
 			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
-			linux,uio-name = "irq_iccom_01_14";
+			linux,uio-name = "irq_iccom_00_22";
 		};
 
-		iccom115 {
+		iccom023 {
 			compatible = "generic-uio";
 			reg = <0 0xe6260000 0 0x10000>;
 			interrupts = <GIC_SPI 368 IRQ_TYPE_LEVEL_HIGH>;
 			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
-			linux,uio-name = "irq_iccom_01_15";
+			linux,uio-name = "irq_iccom_00_23";
 		};
 	};
 
-- 
2.17.1

